tests/suites/external_service/external_service_test.go

```diff
@@ -0,0 +1,138 @@
+)
+
+var (
+	TestName            = "ext-service-test"
+	TestOperatorVersion = "99.99.99-testing"
+	OperatorVersion     string
+	OperatorName        = os.Getenv("OPERATOR_NAME")
+	TestNamespace       = fmt.Sprintf("%s-namespace", TestName)
+	TestInstance        = fmt.Sprintf("%s-instance", OperatorName)
+	KubeConfigPath      = os.Getenv("KUBECONFIG")
+	KubectlPath         = os.Getenv("KUBECTL_PATH")
+	OperatorDirectory   = os.Getenv("OPERATOR_DIRECTORY")
+
+	NodeCount = 1
+	Client    = client.Client{}
+	Operator  = kudo.Operator{}
+)
+
+var _ = BeforeSuite(func() {
+	Client, _ = client.NewForConfig(KubeConfigPath)
+	kubernetes.CreateNamespace(Client, TestNamespace)
+})
+
+var _ = AfterSuite(func() {
+	Operator.Uninstall()
+	kubernetes.DeleteNamespace(Client, TestNamespace)
+})
+
+//var _ = BeforeSuite(func() {
+//	k8s.Init(KubectlOptions)
+//	kudo.Init(KubectlOptions)
+//	_ = kudo.UninstallOperator(OperatorName, TestNamespace, TestInstance)
+//	_ = k8s.CreateNamespace(TestNamespace)
+//})
+//
+//var _ = AfterSuite(func() {
+//	_ = kudo.UninstallOperator(OperatorName, TestNamespace, TestInstance)
+//	_ = k8s.DeleteNamespace(TestNamespace)
+//	if OperatorVersion != "" {
+//		_, _, err := kudo.OverrideOperatorVersion(OperatorVersion)
+//		if err != nil {
+//			log.Errorf(
+//				"Error reverting operatorVersion from '%s' to '%s': %v",
+//				TestOperatorVersion, OperatorVersion, err,
+//			)
+//		}
+//	}
+//})
+
+func TestService(t *testing.T) {
+	RegisterFailHandler(Fail)
+	junitReporter := reporters.NewJUnitReporter(fmt.Sprintf(
+		"%s-junit.xml", TestName,
+	))
+	RunSpecsWithDefaultAndCustomReporters(t, TestName, []Reporter{junitReporter})
+}
+
+func assertNumberOfCassandraNodes(nodeCount int) {
+	nodes, err := cassandra.Nodes(Client, Operator.Instance)
+	Expect(err).To(BeNil())
+	Expect(len(nodes)).To(Equal(nodeCount))
+}
+
+var _ = Describe("external service", func() {
+
+	It("Installs the operator from the current directory", func() {
+		var err error
+
+		Operator, err = kudo.InstallOperator(OperatorDirectory).
+			WithNamespace(TestNamespace).
+			WithInstance(TestInstance).
+			WithParameters(map[string]string{
+				"NODE_COUNT": strconv.Itoa(NodeCount),
+			}).
+			Do(Client)
+		Expect(err).To(BeNil())
+
+		err = Operator.Instance.WaitForPlanInProgress("deploy")
+		Expect(err).To(BeNil())
+
+		err = Operator.Instance.WaitForPlanComplete("deploy")
+		Expect(err).To(BeNil())
+
+		assertNumberOfCassandraNodes(NodeCount)
+	})
+
+	It("Allows external access to the cassandra cluster", func() {
+		err := Operator.Instance.UpdateParameters(map[string]string{
+			"EXTERNAL_NATIVE_TRANSPORT": "true",
+		})
+		Expect(err).To(BeNil())
+
+		assertNumberOfCassandraNodes(NodeCount)
+
+		log.Infof("Verify that external service is started and has 1 open port")
+		svc, err := kubernetes.GetService(Client, fmt.Sprintf("%s-svc-external", TestInstance), TestNamespace)
+		Expect(err).To(BeNil())
+		Expect(len(svc.Spec.Ports)).To(Equal(1))
```

Comment:
Can you also test that the port matches the one configured? `svc.Spec.Port[0].Name == "native-transport" && svc.Spec.Port[0].Port == EXTERNAL_NATIVE_TRANSPORT_PORT`

Reply:
Good point.